mlir/include/mlir/Analysis/Presburger/PwExpr.h

```diff
@@ -0,0 +1,33 @@
+#ifndef MLIR_ANALYSIS_PRESBURGER_PW_EXPR_H
+#define MLIR_ANALYSIS_PRESBURGER_PW_EXPR_H
+
+#include "mlir/Analysis/Presburger/Set.h"
+
+namespace mlir {
+class PresburgerPwExpr {
```

Comment:
In the end this should not only support affine expressions, but all kinds of Presburger expressions. So they should be called piecewise Presburger expressions, but I was not sure if this name would be long.

Reply:
> Is there precedent in the MLIR codebase? 

Not that I know of.

Reply:
What kinds of other presburger expressions? Would that functionality be incorporated into the same class?

Reply:
Divisions for example. But I'm not sure if this will be possible in my remaining time.

Reply:
Even divisions are still Quasi-Affine.

The reason why isl calls these expressions affine is that these expressions are modelled as explicit functions rather than relations encoded through a list of constraints.

Reply:
Would it be better to call them piecewise affine expressions then? And how would we shorten that? 'pwAffineExpression', `pwAffineExpr`, ... ?

Reply:
Think about this, I think `PresburgerExpr` could make a lot of sense, as the term 'Expr' stands for the fact that this is an explicit expression (rather than a collection of constraints), and Presburger could be a shorthand for 'PiecewiseQuasiAffine'.

Reply:
Alternatively, following isl more closely, we could use `PiecewiseAffineExpr`. This one might be more helpful if we want to distinguish them from AffineExpr. I personally would not try to make these terms overly short.

Reply:
I like the `PiecewiseAffineExpr` the most, as it highlights the "piecewise" property of the construct. 

We could also go for `PiecewisePresburgerExpr` as this would be the exactest naming.

Reply:
This would also blend in "nicely" with the corresponding operation: `presburger.piecewiseExpr ...`

Reply:
I think `PiecewisePresburgerExpr` does not really make a lot of sense. Presburger expressions can have many pieces, so adding pieces on top does not add anything.

Reply:
Good point, taken this into consideration `PresburgerExpr` would be good. Then we can use `presburger.expr` as the defining operation.